### PR TITLE
V2.1 Helper Role Additions

### LINKED ISSUE #5: /helper-roles
Assigned to @net-tech, including all subcommands.

Reference this issue for PRs.

### PR DESCRIPTION
**Pull request by tm21cy**
See each commit for more details. Adds helper-roles, its functionality, and associated methodology. Tested with various suites of data, including dummy member data.

TM

### CODE REVIEW

**Comment by avasilic** [comment]
Closes #5

File: src/commands/helper-roles.ts [hunk 1]
```diff
@@ -39,30 +45,119 @@ module.exports = {
...
+        break;
+      }
+      case "ping": {
+        let lang = interaction.options.getString("role", true);
+        let users = (
+          (await Query.helpers.getUsers(lang)).data as HelperCollection
+        ).users;
+        if (users.length == 0) {
+          await interaction.reply({
+            content: "There are no helpers for that role!",
```
Comment (avasilic) [inline src/commands/helper-roles.ts]:
```suggestion
            content: "There are no helpers for that role.",
```

File: src/commands/helper-roles.ts [hunk 2]
```diff
@@ -39,30 +45,119 @@ module.exports = {
...
+        let needle = false;
+        const ITERATIONS = 5;
+        let iteration = 0;
+        if (users.length >= 5) {
+          while (!needle) {
+            let shuffleUsers = _.shuffle(users).slice(0, 5);
+            for (let user of shuffleUsers) {
+              let discordUser = await interaction.guild?.members.resolve(user);
+              if (!discordUser) continue;
+              console.log(discordUser.presence?.status);
```
Comment (avasilic) [inline src/commands/helper-roles.ts]:
```suggestion
```

File: src/commands/helper-roles.ts [hunk 3]
```diff
@@ -39,30 +45,118 @@ module.exports = {
...
+            }
+            iteration++;
+            if (iteration >= 5) break;
+          }
+          let sliceidx = userArray.length >= 3 ? 3 : userArray.length;
+          userArray = _.shuffle(userArray).slice(0, sliceidx);
+        } else userArray = users;
+        let content = "";
+        for (let user of userArray) {
+          content += `<@${user}>`;
```
Comment (avasilic) [inline src/commands/helper-roles.ts]:
```suggestion
          content += `<@${user}> `;
```

File: src/commands/helper-roles.ts [hunk 4]
```diff
@@ -39,30 +45,118 @@ module.exports = {
...
+          }
+          let sliceidx = userArray.length >= 3 ? 3 : userArray.length;
+          userArray = _.shuffle(userArray).slice(0, sliceidx);
+        } else userArray = users;
+        let content = "";
+        for (let user of userArray) {
+          content += `<@${user}> `;
+        }
+        await interaction.reply({
+          content,
```
Comment (avasilic) [inline src/commands/helper-roles.ts]:
```suggestion
          content: `**${lang} Helpers:** ${content}`,
```

**Review by avasilic — APPROVED** [review]

**Comment by avasilic** [comment]
@tm21cy Please take a look at the last change I made, does that make sense?

**Comment by tm21cy** [comment]
All looks good.

**Pull request merged**